src/backend/main.ts

```diff
@@ -879,12 +880,20 @@ ipcMain.handle('requestSettings', async (event, appName) => {
   return mapOtherSettings(config)
 })
 
-ipcMain.handle('toggleDXVK', async (event, { winePrefix, winePath, action }) =>
-  DXVK.installRemove(winePrefix, winePath, 'dxvk', action)
+ipcMain.handle('toggleDXVK', async (event, { appName, action }) =>
+  DXVK.installRemove(
+    await GameConfig.get(appName).getSettings(),
+    'dxvk',
+    action
+  )
 )
 
-ipcMain.on('toggleVKD3D', (event, { winePrefix, winePath, action }) => {
-  DXVK.installRemove(winePrefix, winePath, 'vkd3d', action)
+ipcMain.on('toggleVKD3D', (event, { appName, action }) => {
+  GameConfig.get(appName)
+    .getSettings()
+    .then((gameSettings) => {
+      DXVK.installRemove(gameSettings, 'vkd3d', action)
+    })
```

Comment:
This is nitpicky, but maybe make these have the same format (so either use `.then` in both, or `await` in both)? Otherwise, someone looking at this in the future might wonder why they're structured differently

Reply:
the reason they are different is because the toggleDXVK has to return a Promise<Boolean> and DXVK.InstallRemove returns that Promise<Boolean>

Reply:
You could still change either one to use the format of the other. As an example:
```patch
Index: src/backend/main.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/backend/main.ts b/src/backend/main.ts
--- a/src/backend/main.ts	(revision Staged)
+++ b/src/backend/main.ts	(date 1677780018583)
@@ -881,11 +881,9 @@
 })
 
 ipcMain.handle('toggleDXVK', async (event, { appName, action }) =>
-  DXVK.installRemove(
-    await GameConfig.get(appName).getSettings(),
-    'dxvk',
-    action
-  )
+  GameConfig.get(appName)
+    .getSettings()
+    .then((settings) => DXVK.installRemove(settings, 'dxvk', action))
 )
 
 ipcMain.on('toggleVKD3D', (event, { appName, action }) => {
```

Reply:
The patch I've posted there passes `yarn codecheck`. It still returns a `Promise<boolean>`, as it should.

Are you sure you haven't added curly brackets around everything after the `=>`?
```ts
() => false
```
is different from
```ts
() => { false }
```
after all (one returns `false`, the other `void`)

Reply:
> The patch I've posted there passes `yarn codecheck`. It still returns a `Promise<boolean>`, as it should.
> 
> Are you sure you haven't added curly brackets around everything after the `=>`?
> 
> ```ts
> () => false
> ```
> 
> is different from
> 
> ```ts
> () => { false }
> ```
> 
> after all (one returns `false`, the other `void`)

thanks for that, fixed